src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -0,0 +1,382 @@
+        + ","
+        + "\"values\": [";
+
+    return payload;
+  }
+
+  /**
+   * This constructs a new data point omf message
+   *
+   * @return return the JSON payload for a new OMF data point
+   */
+  public static String addPointToOMFDataMessage(
+      String tagValue, String timestamp, boolean isFirstPoint) {
+    String payload = "";
+
+    if (!isFirstPoint) {
+      payload += ",";
+    }
+
+    payload += "{"
+        + " \"timestamp\": \""
+        + timestamp
+        + ".000Z\","
+        + " \"tagValue\": \"" + tagValue + "\""
+        + "}";
+
+    return payload;
+  }
+
+  /**
+   * call after finished adding all data points to close values array in json
+   *
+   * @return return the JSON payload for the end of an OMF container message.
+   */
+  public static String addContainerEndToOMFDataMessage() {
+    String payload =
+        "]}"; // closes off the end of the container specific portion of the data message
+    return payload;
+  }
+
+  /**
+   * call this when finished building up the OMF message
+   *
+   * @return return the JSON payload for the end of an OMF data message.
+   */
+  public static String endOMFDataMessage() {
+
+    String payload = "]";
+    return payload;
+  }
+
+  /**
+   * Setup the format for how data from this device will be stored with omf
+   *
+   * @return returns the JSON payload to construct a new Type message.
+   */
+  public static String getTypeBody() {
+
+    String flexyName = getFlexyName();
+    String payload = "[{"
+        + "\"id\": \"HMS-type-" + flexyName + "\","
+        + "\"classification\": \"dynamic\","
+        + "\"type\": \"object\","
+        + "\"properties\": {"
+        + "\"timestamp\": {"
+        + "\"type\": \"string\","
+        + "\"format\": \"date-time\","
+        + "\"isindex\": true"
+        + "},"
+        + "\"tagValue\": {"
+        + "\"type\": \"string\","
+        + "\"description\": \"Ewon Flexy's tag value stored as a string\""
+        + "}"
+        + "}"
+        + "}]";
+
+    return payload;
+  }
+
+  /**
+   * should be called for every tag in config list at constructor This sets up OMF to have the
+   * containers
+   */
+  public static String getContainerSettingJson() {
+    String flexyName = getFlexyName();
+    String typeID = "HMS-type-" + flexyName;
+    String payload = startOMFDataMessage();
+
+    for (int i = 0; i < TagInfoManager.getTagInfoList().size(); i++) {
+
+      String tagName = ((TagInfo) TagInfoManager.getTagInfoList().get(i)).getName();
+
+      // after the first tag, separate by comma
+      if (i > 0) {
+        payload += ",";
+      }
+
+      payload += "{"
+          + "\"id\": \"" + tagName + "\","
+          + "\"typeid\": \"" + typeID + "\""
+          + "}";
+    }
+
+    payload += endOMFDataMessage();
+    return payload;
+  }
+
+  /**
+   * Reads the unique name given to the flexy
+   *
+   * @return Name of the Flexy
+   */
+  private static String getFlexyName() {
```

Comment:
I dont know how this is used as a there is a lot of implementation missing here but this should run no more than once to minimize the amount of SCB calls you are doing on a field that wont change.

Reply:
It looks like this is called in both getTypeBody and getContainerSettingsJson.

Reply:
it would be run twice throughout the program. The type and container messages are both sent after we know what tags are being used during startup.